go/systests/team_list_test.go

```diff
@@ -0,0 +1,196 @@
+
+	john := ctx.installKeybaseForUser("john", 10)
+	john.signupNoPUK()
+	t.Logf("Signed up PUK-less user john (%s)", john.username)
+
+	ed := ctx.installKeybaseForUser("ed", 10)
+	ed.signup()
+	ed.reset()
+	ed.loginAfterResetNoPUK(10)
+	t.Logf("Signed up ed (%s), reset, and reprovisioned without PUK", ed.username)
+
+	team := ann.createTeam([]*smuUser{bob, pam})
+	t.Logf("Team created (%s)", team.name)
+
+	pam.reset()
+	t.Logf("Pam resets (%s)", pam.username)
+
+	ann.addWriter(team, john)
+	t.Logf("Adding john (%s)", john.username)
+
+	ann.addWriter(team, ed)
+	t.Logf("Adding ed (%s)", ed.username)
+
+	teamCli := ann.getTeamsClient()
+
+	rootername := randomUser("arbitrary").username
+	_, err := teamCli.TeamAddMember(context.TODO(), keybase1.TeamAddMemberArg{
+		Name:     team.name,
+		Username: rootername + "@rooter",
+		Role:     keybase1.TeamRole_WRITER,
+	})
+	require.NoError(t, err)
+
+	t.Logf("Added rooter (%s@rooter)", rootername)
+
+	// Examine results from TeamGet
+
+	details, err := teamCli.TeamGet(context.TODO(), keybase1.TeamGetArg{
+		Name:        team.name,
+		ForceRepoll: true,
+	})
+	require.NoError(t, err)
+
+	require.Equal(t, 1, len(details.Members.Owners))
+	require.Equal(t, 0, len(details.Members.Admins))
+	require.Equal(t, 4, len(details.Members.Writers))
+	require.Equal(t, 0, len(details.Members.Readers))
+
+	annMember := findMember(ann, details.Members.Owners)
+	require.NotNil(t, annMember)
+	require.True(t, annMember.Active)
+	require.False(t, annMember.NeedsPUK)
+
+	bobMember := findMember(bob, details.Members.Writers)
+	require.NotNil(t, bobMember)
+	require.True(t, bobMember.Active)
+	require.False(t, bobMember.NeedsPUK)
+
+	pamMember := findMember(pam, details.Members.Writers)
+	require.NotNil(t, pamMember)
+	require.False(t, pamMember.Active)
+	require.False(t, pamMember.NeedsPUK)
+
+	johnMember := findMember(john, details.Members.Writers)
+	require.NotNil(t, johnMember)
+	require.True(t, johnMember.Active)
+	require.True(t, johnMember.NeedsPUK)
+
+	edMember := findMember(ed, details.Members.Writers)
+	require.NotNil(t, edMember)
+	require.True(t, edMember.Active)
+	require.True(t, edMember.NeedsPUK)
+
+	require.Equal(t, 1, len(details.AnnotatedActiveInvites))
+	for _, invite := range details.AnnotatedActiveInvites {
+		// There should be only one invite
+		require.EqualValues(t, rootername, invite.Name)
+	}
+
+	// Examine results from TeamList (mostly MemberCount)
+
+	list, err := teamCli.TeamList(context.TODO(), keybase1.TeamListArg{})
+	require.NoError(t, err)
+
+	require.Equal(t, 1, len(list.Teams))
+	require.Equal(t, 0, len(list.AnnotatedActiveInvites))
+
+	teamInfo := list.Teams[0]
+	require.Equal(t, team.name, teamInfo.FqName)
+	require.Equal(t, 5, teamInfo.MemberCount)
+}
+
+func TestTeamDuplicateUIDList(t *testing.T) {
+	tt := newTeamTester(t)
+	defer tt.cleanup()
+	ctx := newSMUContext(t)
+	defer ctx.cleanup()
+
+	ann := makeUserStandalone(t, "ann", standaloneUserArgs{
```

Comment:
Why standalone?

Reply:
hmm, now that I'm thinking about it, it's not needed. I was aiming for something where `ann` wouldn't automatically add `bob` after he gets PUK (and removing the invite), but this wouldn't happen anyway because of the reset and EldestSeqno difference.

Reply:
fixed now, thanks